mapit/shortcuts.py

```diff
@@ -50,25 +50,24 @@ def output_json(out, code=200):
             out['debug_db_queries'] = connection.queries
         indent = 4
     encoder = GEOS_JSONEncoder(ensure_ascii=False, indent=indent)
-    content = encoder.iterencode(out)
 
-    # We don't want a generator function (iterencode) to be passed to an
-    # HttpResponse, as it won't cache due to its close() function adding it to
-    # an instance attribute.
-    content = map(lambda x: x, content)
+    if type(out) is dict:
+        response = http.JsonResponse(out, encoder=GEOS_JSONEncoder, status=code)
```

Comment:
This loses us the indent on staging, and changes the behaviour on non-ASCII characters - the latter shouldn't matter for anything reading it, of course, and the former can be got around easily enough, but it would be good to remain consistent between the two types of output. I think you could pass in a `json_dumps_params` argument.
`response = http.JsonResponse(out, encoder=GEOS_JSONEncoder, status=code, json_dumps_params=json_dumps_params)`